### PR TITLE
sql: mark trace logs of executed statements as sensitive

### PR DESCRIPTION
**Pull request by petermattis**
Only allow viewing of sensitive trace logs when viewing from localhost.

---

This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/5407)

### CODE REVIEW

**Comment by petermattis** [comment]
screen shot 2016-03-19 at 4 35 00 pm

File: server/admin.go [hunk 1]
```diff
@@ -54,13 +54,16 @@ import (
 )
 
 func init() {
-	// Tweak the authentication logic for the tracing endpoint.
-	// By default it's open for localhost only, but with Docker
-	// we want to get there from anywhere.
+	// Tweak the authentication logic for the tracing endpoint. By default it's
+	// open for localhost only, but with Docker we want to get there from
+	// anywhere. We maintain the default behavior of ony allowing access to
```
Comment (tamird) [inline server/admin.go]: s/ony/only/

Comment (petermattis) [inline server/admin.go]: Done.

**Comment by tamird** [comment]
LGTM

**Comment by bdarnell** [comment]
Localhost-only is still too broad. We should only show these when the viewer is authenticated as root or the user who made the request (and even then it should probably be optional to log the queries at all. 

Why are some of the queries still shown in your screenshot?

**Comment by petermattis** [comment]
For each request we log the statement (sensitive) and the pgtag+count (non-sensitive). The 2 lines allow easy determination of how long the statement took to execute.

We should definitely restrict localhost as well. There is a `TODO` in the code to do this. Are you worried about doing this before Beta?

**Comment by bdarnell** [comment]
Ah, got it. Wasn't obvious that those are pgtags and not full count. 

We don't have to fix this for beta; just something to remember for when we get our security story straightened out.

**Pull request merged**